include/sycl/algorithm/inclusive_scan.hpp

```diff
@@ -181,19 +181,22 @@ void buffer_mapscan(ExecutionPolicy &snp,
 
   //WARNING: nb_work_group is not bounded by max_compute_units
   cl::sycl::buffer<B, 1> scan = { cl::sycl::range<1> { nb_work_group } };
+  cl::sycl::range<1> rng_wg {nb_work_group * nb_work_item};
+  cl::sycl::range<1> rng_wi {nb_work_item};
 
   q.submit([&] (cl::sycl::handler &cgh) {
-    cl::sycl::nd_range<1> rng
-      { cl::sycl::range<1>{nb_work_group * nb_work_item},
-        cl::sycl::range<1>{nb_work_item} };
     auto input =
       input_buffer.template get_access<cl::sycl::access::mode::read>(cgh);
     auto output =
       output_buffer.template get_access<cl::sycl::access::mode::write>(cgh);
 
-    cgh.parallel_for_work_group<class wg>(rng, [=](cl::sycl::group<1> grp) {
+    cl::sycl::accessor<B, 1, cl::sycl::access::mode::read_write,
+                       cl::sycl::access::target::local>
+      scratch { cl::sycl::range<1> { size_per_work_group }, cgh };
+
+    cgh.parallel_for_work_group<class wg>(rng_wg, rng_wi, [=](cl::sycl::group<1> grp) {
```

Comment:
Remain the issue of `class wg` which is not unique.
Any idea from Codeplay on how to do this? I guess there is already a way in the PSTL to generate unique class names...